pyplanet/apps/contrib/brawl_match/__init__.py

```diff
@@ -0,0 +1,206 @@
+			await self.brawl_chat(message)
+			await self.ban_map(player_to_ban)
+		else:
+			maps_string = '$z$fff  -  '.join([(await Map.get_by_uid(map[0])).name for map in self.match_maps])
+			await self.brawl_chat(f'Banning phase over! Maps for this match are:')
+			await self.brawl_chat(f'{maps_string}')
+			await self.init_match()
+
+	async def ban_map(self, player):
+		maps = [map[0] for map in self.match_maps]
+		ban_view = BrawlMapListView(self, maps)
+		await ban_view.display(player=player)
+
+	async def remove_map_from_match(self, map_info):
+		self.match_maps.pop(map_info['index']-1)
+
+	async def init_match(self):
+		await self.await_match_start()
+		self.context.signals.listen(mp_signals.map.map_begin, self.set_settings_next_map)
+
+		random.shuffle(self.match_maps)
+
+		await self.brawl_chat(f'Map order: ')
+		for index, (uid, _) in enumerate(self.match_maps, start=1):
+			map_name = (await Map.get_by_uid(uid)).name
+			await self.brawl_chat(f'[{index}/{len(self.match_maps)}] {map_name}')
+
+		await self.instance.map_manager.set_next_map(await Map.get_by_uid(self.match_maps[0][0]))
+		await self.instance.gbx('NextMap')
+
+	async def await_match_start(self):
+		time.sleep(5)
+		await self.brawl_chat(f'Match will start in {self.TIME_UNTIL_MATCH_PHASE} seconds!')
+		time.sleep(self.TIME_UNTIL_BAN_PHASE / 2)
+		await self.brawl_chat(f'Match will start in {int(self.TIME_UNTIL_MATCH_PHASE/2)} seconds!')
+		time.sleep(self.TIME_UNTIL_BAN_PHASE / 4)
+		await self.brawl_chat(f'Match will start in {int(self.TIME_UNTIL_MATCH_PHASE/4)} seconds!')
+		time.sleep(self.TIME_UNTIL_BAN_PHASE / 4)
+		await self.brawl_chat(f'Match will start now!')
+		time.sleep(self.TIME_UNTIL_NEXT_WALL)
+
+
+	async def stop_match(self, player, *args, **kwargs):
+		for signal, target in self.context.signals.listeners:
+			if target == self.set_settings_next_map:
+				signal.unregister(target)
+		self.match_maps = self.brawl_maps
+		self.match_players = []
+
+	async def update_finish_timeout(self, timeout):
+		settings = await self.instance.mode_manager.get_settings()
+		settings['S_FinishTimeout'] = timeout
+		await self.instance.mode_manager.update_settings(settings)
+
+	async def set_settings_next_map(self, map):
+		settings = await self.instance.mode_manager.get_settings()
+		for index, (uid, timeout) in enumerate(self.match_maps):
+			if uid == map.uid:
+				if index == len(self.match_maps) - 1 and settings['S_WarmUpNb'] == 1:
+					settings['S_WarmUpNb'] = 0
+					await self.instance.mode_manager.update_settings(settings)
+				await self.update_finish_timeout(timeout)
+				await self.instance.map_manager.set_next_map(
+					await Map.get_by_uid(
+						self.match_maps[(index + 1) % len(self.match_maps)][0]
+					)
+				)
+
+	async def brawl_chat(self, message, player=None):
+		if player:
+			await self.instance.chat(f'{self.chat_prefix}' + message, player)
+		else:
+			await self.instance.chat(f'{self.chat_prefix}' + message)
```

Comment:
`await self.instance.chat(f'{self.chat_prefix} {message}', player`
and 
`await self.instance.chat(f'{self.chat_prefix} {message}'`